ee/src/main/java/org/jboss/as/ee/component/deployers/ModuleJndiBindingProcessor.java

```diff
@@ -301,16 +302,44 @@ protected void addJndiBinding(final EEModuleConfiguration module, final BindingC
                     service = (BinderService) controller.getService();
                     if (!equals(service.getSource(), bindingConfiguration.getSource())) {
                         throw EeLogger.ROOT_LOGGER.conflictingBinding(bindingName, bindingConfiguration.getSource());
+                    } else {
```

Comment:
no matter no being able to replicate the original issue, and having to re-evaluate the JNDI leftovers after deploy fail, this change is not good, it assumes this is about a bind for same deployment yet this is wrong. There are for instance use cases where the same global JNDI is made by multiple deployments, which we should not fail and (unlike this change) skip the increase of the binding ref counter.

Reply:
are there any tests for that? if thats a real case, Im fine with ditching this. Nothing fail in TS with his change(AFAIR).